src/main/java/sdmExam/Game.java

```diff
@@ -63,4 +63,6 @@ public boolean isPlayerAbleToMakeAMove(Stone player) {
                 .filter(intersection -> !intersection.isOccupied())
                 .anyMatch(emptyIntersection -> !isIllegalMove(player, emptyIntersection));
     }
+
+    public Board getBoard() { return board;}
```

Comment:
I actually wanted to avoid this, it is safer to keep the board private and do not let direct access from other classes

Reply:
Actually, I tried to do the same but just with game.play I couldn't create a case which one of the player can't play. Always I stacked with the diagonal rule in 3x3 and 2x2 board. That's why I had to add by using the board. If it is really something we shouldn't do, we need to find a way to create a case with play and other methods without board

Reply:
Maybe you can add a protected constructor of Game instead, where you pass a Board's instance. Thus inside the test method, you create a board with the conditions of interest and create a Game's instance with the before mentioned constructor.

Reply:
You are right, I tried to implement, thanks